clang/lib/Parse/ParseTentative.cpp

```diff
@@ -1385,6 +1385,12 @@ Parser::isCXXDeclarationSpecifier(ImplicitTypenameContext AllowImplicitTypename,
     if (!getLangOpts().ObjC && Next.is(tok::identifier))
       return TPResult::True;
 
+    if (Next.is(tok::l_paren) &&
+        Tok.getIdentifierInfo()->hasRevertedTokenIDToIdentifier() &&
+        isRevertibleTypeTrait(Tok.getIdentifierInfo())) {
```

Comment:
You can set the correct `tok::TokenKind` here too since it will be set later anyway when this is re-parsed as an expression

Reply:
I would rather not mutate anything in this function